Providers/Modules/Plugins/Common/plugin/out_oms_diag.rb

```diff
@@ -0,0 +1,111 @@
+module Fluent
+
+  class OutputOMSDiag < BufferedOutput
+
+    Plugin.register_output('out_oms_diag', self)
```

Comment:
@lagalbra Added code now to even update omsagent.conf file. Added unit tests as well.